.github/workflows/main-merge.yml

```
@@ -1,18 +1,31 @@
-# Merges any changes from release/prerelease to main (e.g. servicing changes)
+# See https://github.com/dotnet/arcade/blob/e52018a/Documentation/Maestro/New-Inter-Branch-Merge-Approach.md
 
-name: Flow main to release/dev18.0
+name: Inter-branch merge
 on:
-  schedule:
-    # once a day at 13:00 UTC to cleanup old runs
-    - cron: '0 13 * * *'
+  push:
+    branches:
+      - main
```

I don't think we want to update merge PRs on push. I recommend using an every 3 hours schedule as the old GitHub Merge PR bot would.

```suggestion
  schedule:
    # Create a merge every 3 hours.
    - cron: '0 */3 * * *'
```

---

Sure, we can do that, although that works only for main, if we ever add other flows, they need the `push` trigger (because the arcade script has `$env:GITHUB_REF` hardcoded as the source branch to merge).

---

Good point. All the things I would change, if only I had more time.